src/safeposix/syscalls/net_calls.rs

```diff
@@ -891,114 +1022,178 @@ impl Cage {
-
-                            newsockhandle.localaddr = Some(newaddr);
-                            newsockhandle.remoteaddr = Some(remote_addr.clone());
-
-                            //create socket object for new connected socket
-                            newsockhandle.innersocket = Some(acceptedsock);
-
-                            let possibleunixaddr = Self::swap_unixaddr(&remote_addr.clone());
-                            if let interface::GenSockaddr::Unix(_) = possibleunixaddr {
-                                let pathclone = normpath(convpath(possibleunixaddr.path().clone()), self);
-                                if let Some(inodenum) = metawalk(pathclone.as_path()) {                
-                                    newsockhandle.realdomain = AF_UNIX;
-                                    newsockhandle.unix_info = Some(UnixSocketInfo {
-                                        reallocalpath: pathclone,
-                                        inode: inodenum.clone(),
-                                        mode: sockhandle.unix_info.as_ref().unwrap().mode,
-                                    });
-                                    if let Inode::Socket(ref mut sock) = *(FS_METADATA.inodetable.get_mut(&inodenum).unwrap()) { 
-                                        sock.refcount += 1; 
-                                    } 
-                                };
-                            };
-                            
-                            let _insertval = newfdoption.insert(Socket(newsockfd));
-                            *addr = remote_addr; //populate addr with what address it connected to
+                let remote_addr : interface::GenSockaddr;
+                let sendpipenumber;
+                let receivepipenumber;
 
-                            return newfd;
+                loop {
+                    let localpathbuf = normpath(convpath(sockhandle.localaddr.unwrap().path().clone()), self);
+                    let dsconnobj = NET_METADATA.domsock_accept_table.get(&localpathbuf);
+
+                    if let Some(ds) = dsconnobj {
+                        // if its blocking it will have a connvar, signal it
+                        if let Some(connvar) = ds.get_cond_var() {
+                            if !connvar.signal() { 
+                                drop(ds);
+                                continue;
+                            }
                         }
-                        _ => {
-                            return syscall_error(Errno::EOPNOTSUPP, "accept", "Unkown protocol in accept");
+                        let addr = ds.get_sockaddr().clone();
+                        remote_addr = addr.clone();
+                        receivepipenumber = ds.get_receive_pipe().clone();
+                        sendpipenumber = ds.get_send_pipe().clone();
+                        drop(ds);
+                        NET_METADATA.domsock_accept_table.remove(&localpathbuf);
```

Comment:
Yeah that makes sense.

@yizhuoliang i think we need to use remove instead of get earlier in the code, if theres something there, remove() will return a key/value that we can do these operations with and will avoid the race condition of dropping and removing after